runtime/taskcontext.go

```diff
@@ -68,13 +76,17 @@ type TaskContextController struct {
 }
 
 // NewTaskContext creates a TaskContext and associated TaskContextController
-func NewTaskContext(tempLogFile string) (*TaskContext, *TaskContextController, error) {
+func NewTaskContext(tempLogFile string, claim *queue.TaskClaimResponse) (*TaskContext, *TaskContextController, error) {
```

Comment:
@imbstack would you not wish to avoid having to mock that?
Isn't easier if we just pass in `TaskInfo` ?